src/mewbot/loader.py

```diff
@@ -103,11 +96,25 @@ def load_component(config: ConfigBlock) -> Component:
 
     target_class: Type[Any] = getattr(module, config["name"])
 
-    if not issubclass(target_class, Component):
+    # Verify that the implementation class matches the interface we got from
+    # the `kind:` hint.
+    if not issubclass(target_class, interface):
         raise Exception("Trying to load a non-registered class from config")
 
     component = target_class.__call__(uid=config["uuid"], **config["properties"])
 
-    assert isinstance(component, Component)
+    # Verify the instance implements a valid interface.
```

Comment:
Suggest some kind of error message here for if the assert fails.

Reply:
I'm still not over the fact that python's assert doesn't have `assert([bool], [str])`

Reply:
...

That would be really kinda neat.

I mean, they'll probably reject it, but I'm weirdly game for drafting a PEP

Reply:
It turns out, it does work like that! It's just not obvious because `assert` _doesn't use brackets_.

```python
>>> assert(False, "Hello")
>>> assert False, "Hello"
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError: Hello
```

The first one gets interpreted as `Tuple[Bool=False, Str="Hello"]` which is truthy.

Reply:
Ahhhhhhhhh!

Yes. I, too, am far from a fan!